Assets/Scripts/Weapons/Gun.cs

```diff
@@ -53,7 +53,7 @@ void Update()
             heat -= Time.deltaTime * heatDecayRate;
         }
         
-        if(firing)
+        if(firing && realteam == Teams.playerTeam)

```

Comment:
I'm not sure I like this way of disabling the overheat for enemies. Can we have a public boolean for `overheatEnabled` instead?